internal/cmd/policy/listrules/listrules.go

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package listrules
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/gittuf/gittuf/internal/cmd/common"
+	"github.com/gittuf/gittuf/internal/cmd/policy/persistent"
+	"github.com/gittuf/gittuf/internal/policy"
+	"github.com/gittuf/gittuf/internal/repository"
+	"github.com/spf13/cobra"
+)
+
+type options struct {
+	p *persistent.Options
+}
+
+func (o *options) AddFlags(_ *cobra.Command) {}
+
+func (o *options) Run(cmd *cobra.Command, _ []string) error {
+	repo, err := repository.LoadRepository()
+	if err != nil {
+		return err
+	}
+
+	rules, err := repo.ListRules(cmd.Context())
+	if err != nil {
+		return err
+	}
+
+	// DFS through the rules
+
+	for _, curRule := range rules {
+		if curRule.Delegation.Name == policy.AllowRuleName {
+			continue
+		}
+		fmt.Printf(strings.Repeat("    ", curRule.Depth)+"Rule %s:\n", curRule.Delegation.Name)
+		gitpaths, filepaths := []string{}, []string{}
+		for _, path := range curRule.Delegation.Paths {
+			if strings.HasPrefix(path, "git:") {
+				gitpaths = append(gitpaths, path)
+			} else {
+				filepaths = append(filepaths, path)
+			}
+		}
+		if len(filepaths) > 0 {
+			fmt.Println(strings.Repeat("    ", curRule.Depth+1) + "Paths affected:")
+			for _, v := range filepaths {
+				fmt.Printf(strings.Repeat("    ", curRule.Depth+2)+"%s\n", v)
+			}
+		}
+		if len(gitpaths) > 0 {
+			fmt.Println(strings.Repeat("    ", curRule.Depth+1) + "Refs affected:")
+			for _, v := range gitpaths {
+				fmt.Printf(strings.Repeat("    ", curRule.Depth+2)+"%s\n", v)
+			}
+		}
+
+		fmt.Println(strings.Repeat("    ", curRule.Depth+1) + "Authorized keys:")
+		for _, key := range curRule.Delegation.Role.KeyIDs {
+			fmt.Printf(strings.Repeat("    ", curRule.Depth+2)+"%s\n", key)
+		}
+
+		if curRule.Delegation.Custom != nil {
+			customString, err := json.Marshal(curRule.Delegation.Custom)
+			if err != nil {
+				return fmt.Errorf("error marshalling custom field: %w", err)
+			}
+			fmt.Printf(strings.Repeat("    ", curRule.Depth+1)+"Details: %s\n", string(customString))
+		}
+
+		fmt.Println()
+	}
+	return nil
+}
+
+func New(persistent *persistent.Options) *cobra.Command {
```

Comment:
I think we don't need `persistent` for this one.

Reply:
Removed